### DISCUSSION TITLE
[Question] How to develop isaac sim demo

### DISCUSSION THREAD

**Discussion by MinnaZhong** [isaac-sim/IsaacLab#1452] [Q&A]
Hi Team,

I am Minna from UFactory. Recently we wanted to implement the same demo with our robotic arm(xArm6), but we didn't know how to configure it. We can do simple movement, and already generated USD file, you can download the file in the link below.
Do you need to configure this from the background? Kindly give us some guidance, thank you in advance.

USD file:
https://drive.google.com/drive/folders/1TEAFwETfudnYOz773uX2a3_cVOpQBkdp?usp=sharing

https://docs.omniverse.nvidia.com/isaacsim/latest/features/environment_setup/assets/usd_assets_robots.html
![image](https://github.com/user-attachments/assets/43877f46-b8e1-4a5c-8f0d-a5d19db01653)

Best regards,
Minna

**Comment by mpgussert** [comment]
Hello @MinnaZhong !

Sorry for the delayed response in getting back to you! Unfortunately there's little I can do to help with the information you have provided me, other than to assure you that we will better document this process.

When you say that you can do "simple movement and already generated a USD" do you mean that your USD has an articulation setup and you can control it? Are you trying to just tune the joints appropriately?  Or is the issue with controlling it programmatically or with a trained policy like in the demo? What is "simple movement" specifically? Please provide more information so we can help!

Thanks!
Gus

**Comment by penglongxiang** [comment]
Hi @mpgussert, thanks for your kind reply. I am also a developer from [UFACTORY](https://github.com/xArm-Developer), we are a robotics product provider and have a large customer base, as requested by our users, as well as part of our development plan, we would like to ask is there a way to request an **official integration support** for our robot models (provided as usd files in links from the initial post), just like Franka, Kinova and Universal Robots, already integrated in Isaac Sim [robot assets](https://docs.omniverse.nvidia.com/isaacsim/latest/features/environment_setup/assets/usd_assets_robots.html). 

We have already tuned the joint drives of our usd models, and already tested by articulation position controllers, this is what "simple movement" means in our previous post. Next we plan to move on to RMPFlow configuration and motion planning demos. We wonder in order to complete an Isaac sim integration (if possible), what other stuff we need to provide and is there a guidance for this purpose? 

In short, we can do the development ourselves, just curious about how to make our models officially integrated in Isaac sim assets, demos and docs, making it easy for our users who are interested in RL to simulate our manipulators in your great platform. Thank you in advance, looking forward to your reply.

**Reply by fl1ps1de5** [reply]
Hi there - I was wondering if you guys at UFACTORY have any pre-configured environments/tasks written for IsaacLab/Sim that utilise the xarm6

Such as a reaching, gripping task.

I currently am working in RL, on some some tasks with the Franka arm, but seeing as my lab only has the xarm6 in real life, I want to simulate this too (so I can eventually run the tasks on the real arm)

Thanks a lot

**Comment by RandomOakForest** [comment]
Thank you for posting this. Apologies for the delayed response. We'll move this into our _Discussions_ section for the team to follow up. Thanks.
